src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -100,9 +102,10 @@ rtypePredBinds = map uPVar . ty_preds . toRTypeRep
 ofBRType :: (PPrint r, Reftable r)
          => (SourcePos -> RTAlias RTyVar SpecType -> [BRType r] -> r -> BareM (RRType r))
          -> (r -> BareM r)
+         -> (Expr -> BareM Expr)
          -> BRType r
          -> BareM (RRType r)
-ofBRType appRTAlias resolveReft
+ofBRType appRTAlias resolveReft resolveExpr
```

Comment:
Looks to me like this extra parameter is not needed, why not just inline the `resolve <=< expandExpr` bit?